core/services/ocr2/plugins/ocr2keeper/custom_telemetry.go

```diff
@@ -0,0 +1,159 @@
+package ocr2keeper
+
+import (
+	"context"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+
+	httypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/headtracker/types"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/keeper_registry_wrapper2_0"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/static"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+const (
+	customTelemChanSize = 100
+)
+
+type AutomationCustomTelemetryService struct {
+	utils.StartStopOnce
+	monitoringEndpoint commontypes.MonitoringEndpoint
+	headBroadcaster    httypes.HeadBroadcaster
+	headCh             chan blockKey
+	unsubscribe        func()
+	chDone             chan struct{}
+	lggr               logger.Logger
+	configDigest       [32]byte
+	latestConfigDigest latestConfigDigest
+}
+
+type latestConfigDigest interface {
+	LatestConfigDetails(opts *bind.CallOpts) (keeper_registry_wrapper2_0.LatestConfigDetails, error)
+}
+
+// NewAutomationCustomTelemetryService creates a telemetry service for new blocks and node version
+func NewAutomationCustomTelemetryService(me commontypes.MonitoringEndpoint, hb httypes.HeadBroadcaster, lggr logger.Logger, cd [32]byte, latestcd latestConfigDigest) *AutomationCustomTelemetryService {
+	return &AutomationCustomTelemetryService{
+		monitoringEndpoint: me,
+		headBroadcaster:    hb,
+		headCh:             make(chan blockKey, customTelemChanSize),
+		chDone:             make(chan struct{}),
+		lggr:               lggr.Named("Automation Custom Telem"),
+		//HERE initailize
+		configDigest:       cd,
+		latestConfigDigest: latestcd,
+	}
+}
+
+// Start starts Custom Telemetry Service, sends 1 NodeVersion message to endpoint at start and sends new BlockNumber messages
+func (e *AutomationCustomTelemetryService) Start(ctx context.Context) error {
+	return e.StartOnce("AutomationCustomTelemetryService", func() error {
+		e.lggr.Infof("Starting: Custom Telemetry Service")
+		// build call Opts
+		go func() {
+			ticker := time.NewTicker(1 * time.Minute)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					callOpt := &bind.CallOpts{Context: ctx, BlockNumber: nil}
+					newConfigDetails, cdErr := e.latestConfigDigest.LatestConfigDetails(callOpt)
+					e.lggr.Errorf("Error occurred while getting newestConfigDetails  %v", cdErr)
+					newConfigDigest := newConfigDetails.ConfigDigest
+					if newConfigDigest != e.configDigest {
```

Comment:
no need to check here, just assign the new config digest to `e.configDigest`

Reply:
is there a reason this is preferred?

Reply:
i think this adds more for readers. but no strong preference here.